lib/zendesk_api/collection.rb

```diff
@@ -410,11 +441,32 @@ def get_response(path)
       end
     end
 
+    def handle_cursor_response(response_body)
+      unless response_body.is_a?(Hash)
+        raise ZendeskAPI::Error::NetworkError, @response.env
+      end
+
+      response_body = get_next_page_data(response_body) if has_more_results?(response_body)
+
+      body = response_body.dup
+      results = body.delete(@resource_class.model_key) || body.delete("results")
+
+      unless results
+        raise ZendeskAPI::Error::ClientError, "Expected #{@resource_class.model_key} or 'results' in response keys: #{body.keys.inspect}"
+      end
+
+      @resources = results.map do |res|
+        wrap_resource(res)
+      end
+    end
+
     def handle_response(response_body)
       unless response_body.is_a?(Hash)
         raise ZendeskAPI::Error::NetworkError, @response.env
       end
 
+      response_body = get_next_page_data(response_body) if has_more_results?(response_body)
```

Comment:
Why do we need this in the original method?

Reply:
Yep, that is not needed.